compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -686,23 +686,49 @@ impl<'a> Parser<'a> {
         }
 
         if let token::DocComment(kind, style, _) = self.token.kind {
-            // We have something like `expr //!val` where the user likely meant `expr // !val`
-            let pos = self.token.span.lo() + BytePos(2);
-            let span = self.token.span.with_lo(pos).with_hi(pos);
-            err.span_suggestion_verbose(
-                span,
-                format!(
-                    "add a space before {} to write a regular comment",
-                    match (kind, style) {
-                        (token::CommentKind::Line, ast::AttrStyle::Inner) => "`!`",
-                        (token::CommentKind::Block, ast::AttrStyle::Inner) => "`!`",
-                        (token::CommentKind::Line, ast::AttrStyle::Outer) => "the last `/`",
-                        (token::CommentKind::Block, ast::AttrStyle::Outer) => "the last `*`",
-                    },
-                ),
-                " ".to_string(),
-                Applicability::MachineApplicable,
-            );
+            // This is used to avoid adding suggestions for doc comments in item list.
+            // For example,
+            // `enum Foo {
+            //     Bar ///baz
+            //     Baz }`
+            // should suggest to add a comma before `///baz`.
+            let is_expect_item_list_comma =
+                edible.len() == 1 && edible.iter().any(|e| e.token_type == TokenType::Comma);
+            if !is_expect_item_list_comma {
```

Comment:
It doesn't have to be an item list, it can be any list as the review I made previously
https://github.com/rust-lang/rust/pull/142341#pullrequestreview-2919666177